packages/analytics/analytics-chart/src/utils/commonOptions.ts

```diff
@@ -1,16 +1,14 @@
 import type { ExternalTooltipContext, KChartData, TooltipState, TooltipEntry, Dataset, ChartLegendSortFn, LegendValues, EnhancedLegendItem, TooltipInteractionMode } from '../types'
-import { formatUnit } from '../utils'
+import { formatByGranularity, formatUnit } from '../utils'
 import { isValid } from 'date-fns'
 import type { Chart, Point, ScatterDataPoint } from 'chart.js'
-import { formatTime } from '@kong-ui-public/analytics-utilities'
+import type { GranularityValues } from '@kong-ui-public/analytics-utilities'
 
 export const isTooltipInteractive = (state: TooltipInteractionMode) => {
   return ['interactive', 'zoom-interactive'].includes(state)
 }
 
-// TODO: we should implement a separate tooltip behavior for each broad chart type
-// as the "tooltip behaviors" are beggining to diverge more across chart types.
-export const tooltipBehavior = (tooltipData: TooltipState, context: ExternalTooltipContext) : void => {
+export const lineChartTooltipBehaviour = (tooltipData: TooltipState, context: ExternalTooltipContext, granularity: GranularityValues) : void => {
```

Comment:
nit: This is a silly one, but we should be consistent with our locality when naming these things. `Behaviour` (UK spelling) vs. `Behavior` (US spelling).

Since `tooltipBehavior` is defined below, we should go with that spelling.